docs/DeveloperGuide.md

```diff
@@ -1,34 +1,678 @@
+
+The `addNotes` command first starts with 2 paths: with or without title. If a title is added alongside the 
+“addNotes” (ie. the user inputs “addNotes cats”) then the title is passed on to a method `isValidTitle` that would 
+check for a blank input. Conversely, if no title is appended, then the system would prompt the user for the title. 
+Either way, the system would still check for if the user inputs an empty title. 
+Next, the system would prompt the user for a description. Similar to title, the same method `isValidTitle`, to check 
+if the user has inputted an empty String.
+Finally, if both title and description are of valid input, then the 2 parameters would be added to the `NoteList` 
+and passed on to `NoteStorage`. 
+
+Figure 4.4-1 below illustrates the flow of `addNotes` through a sequence diagram.
+
+![Figure 4.4-1](images/DeveloperGuide/Figure 4.4-1.png)
+_Figure 4.4-1: Sequence diagram for addNotesCommand_
+
+
+##### Design Considerations
+
+An empty string must be defined clearly (a long string of spaces/no string) and must not be inputted 
+by the user as it affects the usability and searchability of the `NoteList`. As such, checks would need to
+be implemented to prevent any hiccups by the user.
+
+
+### 4.5 Editing and Deleting Notes (Edmund)
+
+The `editNotes` command allows the user to change the title or description of their notes stored in the `NoteList`.
+The `deleteNotes` command allows the user to remove the specified notes completely from the `NoteList`.
+
+##### Implementation
+
+Below illustrates the steps taken by each command to ensure their execution.
+
+###### editNotes
+
+1. `checkEmptyList` would be called to ensure the `NoteList` is not empty before proceeding on with the execution. 
+If an empty list is detected, the command would terminate with a prompt of empty list to the user.
+1. A condition of whether the user has pre-inputted the title is checked.
+    * If the user pre-inputs a title, the findTitle function would go through the list to find the title. If the input 
+    title does not match any of the titles in the list, the command would terminate with a message: 
+    “title is not found”. If a match is found, the system prints out the note and asks for which part to edit. 
+    In the case of multiple matches, the system would print all matching cases and ask the user to select amongst them. 
+    Following the confirmation, a “T” would mean editing the title while “D” would mean editing the description.
+    * If the user did not pre-input the title, the system would print all notes currently in the list and ask for 
+the user to select which note to edit among them. Following the confirmation, a “T” would mean editing the title 
+while “D” would mean editing the description.
+       
+1. The new edited note would then be passed on to storage for saving.
+
+Figure 4.5-1 illustrates the above steps via a sequence diagram.
+
+![Figure 4.5-1](images/DeveloperGuide/Figure 4.5-1.png)
```

Comment:
![image](https://user-images.githubusercontent.com/57716189/97389416-2faff380-1915-11eb-985a-4199212bdd18.png)

i think this should be the case?

Reply:
I agree with @Speedweener 

![image](https://user-images.githubusercontent.com/60292763/97657383-adeed000-1aa4-11eb-9d7b-ccf823260180.png)

The return values from self-called methods are missing in this diagram also. And perhaps you should add dashed line  from `:Note` to `:EditNotesCommand` to give back the control to `:EditNotesCommand`